cluster/assemblestate/assemblestate_test.go

```diff
@@ -174,6 +174,13 @@ func (s *assembleSuite) TestRun(c *check.C) {
 	c.Assert(routes.Addresses, testutil.DeepUnsortedMatches, addrs)
 	c.Assert(routes.Devices, testutil.DeepUnsortedMatches, rdts)
 	c.Assert(len(routes.Routes)/3, check.Equals, count*(count-1))
+
+	c.Assert(ids, check.HasLen, count)
```

Comment:
This gets checked implicitly by the unsorted matching btw. Doesn't hurt to have it anyway of course